AWSAppSyncClient/Internal/AWSMutationCache.swift

```diff
@@ -0,0 +1,153 @@
+//
+// Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// You may not use this file except in compliance with the License.
+// A copy of the License is located at
+//
+// http://aws.amazon.com/apache2.0
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+//
+
+import Foundation
+import SQLite
+
+public final class AWSMutationCache {
+
+    private let db: Connection
+    private let mutationRecords = Table("mutation_records")
+    private let id = Expression<Int64>("_id")
+    private let recordIdentifier = Expression<CacheKey>("recordIdentifier")
+    private let data = Expression<Data>("data")
+    private let recordState = Expression<String>("recordState")
+    private let timestamp = Expression<Date>("timestamp")
+    private let s3Bucket = Expression<String?>("s3Bucket")
```

Comment:
With some customer asks about supporting more than 1 s3 file/ complex object per mutation, do you think it is a good idea to use this opportunity to change the schema to hold an array of s3 objects?

Reply:
I'd like to put that in a separate change, since this PR is a) targeted toward refactoring database handling; and b) quite large enough. :)